daemon/graphdriver/driver.go

```diff
@@ -114,30 +116,37 @@ func New(root string, options []string) (driver Driver, err error) {
 
 	// Guess for prior driver
 	priorDrivers := scanPriorDrivers(root)
+
+	_, ok := priorDrivers["devicemapper"]
+	// Do not allow devicemapper when it's not explicit and the Docker binary was built statically.
+	if ok && len(DefaultDriver) == 0 && dockerversion.IAMSTATIC == "true" {
```

Comment:
Isn't `len(DefaultDriver) == 0` necessarily true in this code path? It seems that if it wasn't, we would have returned above in line 113, no?